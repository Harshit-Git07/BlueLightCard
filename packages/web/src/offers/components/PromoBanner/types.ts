export interface PromoBannerProps {
  image: string;
  href: string;
}
