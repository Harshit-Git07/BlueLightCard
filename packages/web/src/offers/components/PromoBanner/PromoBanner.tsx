import Link from '@/components/Link/Link';
import Image from '@/components/Image/Image';
import { PromoBannerProps } from './types';

const PromoBanner = ({ image, href, id }: PromoBannerProps) => {
  return (
    <div className="w-full relative">
      <Link href={href} useLegacyRouting data-testid={id}>
        <Image
          src={image}
          alt="Banner image"
          fill={false}
          width="0"
          height="0"
          sizes="100vw"
          className={'h-auto w-full'}
          quality={50}
        />
      </Link>
    </div>
  );
};

export default PromoBanner;
