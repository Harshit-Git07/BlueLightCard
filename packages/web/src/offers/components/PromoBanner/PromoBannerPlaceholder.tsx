import React from 'react';
import ContentLoader from 'react-content-loader';

function PromoBannerPlaceholder() {
  return (
    <div className="w-full relative mb-2">
      <ContentLoader
        speed={3}
        viewBox="0 0 600 200"
        backgroundColor="#f3f3f3"
        foregroundColor="#d1d1d1"
      >
        <rect x="0" y="0" rx="0" ry="0" width="600" height="200" />
      </ContentLoader>
    </div>
  );
}

export default PromoBannerPlaceholder;
