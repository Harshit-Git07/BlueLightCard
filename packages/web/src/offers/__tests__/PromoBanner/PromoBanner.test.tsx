import PromoBanner from '@/offers/components/PromoBanner/PromoBanner';
import renderer from 'react-test-renderer';

describe('Promo Banner', () => {
  it('should render with the correct link and image url', () => {
    const component = renderer.create(<PromoBanner image="test.com" href="test.com" />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
