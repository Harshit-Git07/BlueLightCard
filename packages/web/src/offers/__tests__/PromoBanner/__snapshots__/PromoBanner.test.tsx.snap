// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Promo Banner should render with the correct link and image url 1`] = `
<div
  className="w-full relative"
>
  <a
    data-testid="anchor-link"
    href="test.com"
  >
    <img
      alt="Banner image"
      className="h-auto w-full"
      data-nimg="1"
      decoding="async"
      height={0}
      loading="lazy"
      onError={[Function]}
      onLoad={[Function]}
      sizes="100vw"
      src="_next/static/test.com?width=3840&quality=50"
      srcSet="_next/static/test.com?width=640&quality=50 640w, _next/static/test.com?width=750&quality=50 750w, _next/static/test.com?width=828&quality=50 828w, _next/static/test.com?width=1080&quality=50 1080w, _next/static/test.com?width=1200&quality=50 1200w, _next/static/test.com?width=1920&quality=50 1920w, _next/static/test.com?width=2048&quality=50 2048w, _next/static/test.com?width=3840&quality=50 3840w"
      style={
        {
          "backgroundImage": "url("data:image/svg+xml;charset=utf-8,%3Csvg xmlns='http%3A//www.w3.org/2000/svg'%3E%3Cimage style='filter:blur(20px)' preserveAspectRatio='none' x='0' y='0' height='100%25' width='100%25' href='data:image/png;base64,iVBORw0KGgoAAAANSUhEUgAAAAEAAAABCAYAAAAfFcSJAAAADUlEQVR42mPs7+9bAQAFtwJW4mO4owAAAABJRU5ErkJggg=='/%3E%3C/svg%3E")",
          "backgroundPosition": "50% 50%",
          "backgroundRepeat": "no-repeat",
          "backgroundSize": "cover",
          "color": "transparent",
        }
      }
      width={0}
    />
  </a>
</div>
`;
